stratum/p4c_backends/fpm/Dockerfile

```diff
@@ -14,36 +14,23 @@
 # limitations under the License.
 #
 
-# This Dockerfile expects the stratum root as its scope, hence you should build
-# from root e.g.:
-# docker build -t <some tag> -f stratum/p4c_backends/fpm/Dockerfile .
+# This Dockerfile expects its containing directory as its scope with the p4c-fpm
+# Debian package present, hence you should build it like this:
+# bazel build //stratum/p4c_backends/fpm:p4c_fpm_deb
+# cp bazel-bin/stratum/p4c_backends/fpm/p4c_fpm_deb.deb stratum/p4c_backends/fpm
+# cd stratum/p4c_backends/fpm
+# docker build -t <some tag> -f Dockerfile  .
 
-# We use a 2-stage build.
-FROM stratumproject/build:build as builder
-
-# Build FPM backend
-COPY . /tmp/stratum
-WORKDIR /tmp/stratum
-
-RUN bazel build //stratum/p4c_backends/fpm:p4c-fpm
-
-# Put aside binary and other runtime dependencies.
-RUN mkdir -p /output/usr/local/bin
-RUN install bazel-bin/stratum/p4c_backends/fpm/p4c-fpm /output/usr/local/bin/
-
-RUN mkdir -p /output/usr/share/p4c
-RUN cp -r $(bazel info output_base)/external/com_github_p4lang_p4c/p4include /output/usr/share/p4c/
-
-# Final stage, runtime.
-FROM bitnami/minideb:stretch as runtime
+FROM bitnami/minideb:stretch
 
 LABEL maintainer="Stratum dev <stratum-dev@lists.stratumproject.org>"
 LABEL description="Docker-based distribution of the Stratum p4c FPM backend"
 
 ENV RUNTIME_DEPS libgmp10 libgmpxx4ldbl gcc
 RUN install_packages ${RUNTIME_DEPS}
 
-COPY --from=builder /output /
+ADD ./p4c_fpm_deb.deb /
```

Comment:
The resulting docker image will have the deb package taking up space unnecessarily. Before, only the p4c-fpm binary and dependencies were copied to the final image. You might have simplified the Dockerfile, but image efficiency seems to have dropped. I don't know how big is the debian package, so I'm not sure whether such inefficiency can be ignored. You could use the same approach as before, i.e., install the deb package in the first stage and copy over the binary in the second stage.

Reply:
Done. Should I add the comments back in too?

Reply:
Up to you. I think it should be pretty straightforward for anyone with knowledge of docker multi-stage builds.